src/plugins/discover/public/application/apps/main/utils/update_search_source.ts

```diff
@@ -19,19 +20,15 @@ export function updateSearchSource({
   indexPattern,
   services,
   sort,
-  columns,
   useNewFieldsApi,
-  showUnmappedFields,
   persistentSearchSource,
   volatileSearchSource,
 }: {
   indexPattern: IndexPattern;
   services: DiscoverServices;
   sort: SortOrder[];
-  columns: string[];
   useNewFieldsApi: boolean;
-  showUnmappedFields?: boolean;
-  persistentSearchSource: ISearchSource;
+  persistentSearchSource?: ISearchSource;
   volatileSearchSource?: ISearchSource;
```

Comment:
@timroes I would suggest to further simplify here, since `volatileSearchSource` and `persistentSearchSource` have the same type. It'd suggest to provide 1 searchSource and 1 boolean to determine how it should be treated. simpler typings, bit simpler code. wdyt?

```
{
  indexPattern: IndexPattern;
  services: DiscoverServices;
  sort: SortOrder[];
  useNewFieldsApi: boolean;
  searchSource: ISearchSource;
  isPersisted: boolean;
}
```

Reply:
I am fine with either option.

Reply:
refactored this